src/StreamJsonRpc/Reflection/MessageFormatterRpcMarshaledContextTracker.cs

```diff
@@ -210,16 +227,37 @@ internal MarshalToken GetToken(IRpcMarshaledContext<object> marshaledContext)
             throw new NotSupportedException("Receiving marshaled objects scoped to the lifetime of a single RPC request is not yet supported.");
         }
 
+        Type actualInterfaceType = interfaceType;
+        if (token.Value.SubTypeCode.HasValue)
+        {
+            Type? matchedSubType = null;
+            foreach (RpcMarshalableKnownSubtypeAttribute attribute in interfaceType.GetCustomAttributes(typeof(RpcMarshalableKnownSubtypeAttribute), inherit: false))
+            {
+                if (attribute.SubTypeCode == token.Value.SubTypeCode.Value)
+                {
+                    matchedSubType = attribute.SubType;
```

Comment:
I should probably add a `throw` here if `attribute.SubType` doesn't extend `interfaceType` (otherwise it will fail client-side when casting).

Are there other conditions that we want to check?
We could verify that there aren't duplicated `SubTypeCode` values but it would make this code more complicated and slower.

Reply:
I don't think we should throw. That doesn't lend well to duck typing and versioning being independent between the two parties. Rather, I think we should implement every interface that _does_ match, including the base interface itself (which doesn't appear in the list, but is assumed to be a given because we require subtypes to derive from it.)

Reply:
As discussed in person, I won't add multiple interfaces to the proxy because it may result in confusing situations if the different interfaces define the methods with the same name that are implemented separately with explicit implementations.
I have updated my proposed additions to the protocol to allow multiple subtype codes though, so that we can add this feature in the future,